adc/onboard_temperature/onboard_temperature.c

```diff
@@ -0,0 +1,48 @@
+#include <stdio.h>
+#include "pico/stdlib.h"
+#include "hardware/adc.h"
+
+/* References for this implementation:
+ * raspberry-pi-pico-c-sdk.pdf, page 68. 
+ * pico-examples/adc/adc_console/adc_console.c */
+const float read_onboard_temperature(const char unit) {
+    
+    /* 12-bit conversion, assume max value == ADC_VREF == 3.3 V */
+    const float conversionFactor = 3.3f / (1 << 12);
+
+    adc_select_input(4);
```

Comment:
This only needs to be done once (you're not changing ADC inputs) so could be moved to `main`?

Reply:
I thought a little bit about this. Efficiency-wise, yes, it would be better move it to `main`. But I have at least two points in favor of letting it inside `read_onboard_temperature`:
* It looks more concise in terms of scope to put it in the function;
* `adc_select_input()` is a global operation, so something else along a code (not this one) may be changing it.

But that's OK I can follow yout suggestion. For the sake of this example, is OK. ;)